File: customer-centric-website/frontend/src/api/booking.ts
import api from "./axios";
import type { BookingFormData } from "../types/Booking";

export const submitBooking = async (data: BookingFormData) => {
  const formattedData = {
    full_name: data.fullName,
    company_name: data.companyName,
    email: data.email,
    phone: data.phone,
    budget: data.budget,
    date: data.date,
    time: data.time,
    notes: data.notes,

    slot_id: data.slotId,   // ✅ Added
  };

  const response = await api.post("/bookings", formattedData);

  return response.data;
};

export const getBookings = async () => {
  const response = await api.get("/bookings");
  return response.data;
};

export const deleteBooking = async (id: number) => {
  const response = await api.delete(`/bookings/${id}`);
  return response.data;
};

File: customer-centric-website/backend/models/Booking.go
package models

import "time"

type Booking struct {
	ID uint `gorm:"primaryKey"`

	FullName string `json:"full_name"`

	CompanyName string `json:"company_name"`

	Email string `json:"email"`

	Phone string `json:"phone"`

	Budget string `json:"budget"`

	Date string `json:"date"`

	Time string `json:"time"`

	Notes string `json:"notes"`


	SlotID uint `json:"slot_id"`

	CreatedAt time.Time
}